src/storage/v3/shard_rsm.cpp

```diff
@@ -510,9 +514,125 @@ msgs::WriteResponses ShardRsm::ApplyWrite(msgs::CommitRequest &&req) {
   return msgs::CommitResponse{};
 };
 
-// NOLINTNEXTLINE(readability-convert-member-functions-to-static)
-msgs::ReadResponses ShardRsm::HandleRead(msgs::GetPropertiesRequest && /*req*/) {
-  return msgs::GetPropertiesResponse{};
+msgs::ReadResponses ShardRsm::HandleRead(msgs::GetPropertiesRequest &&req) {
+  if (req.vertices_and_edges.empty()) {
+    return msgs::GetPropertiesResponse{.result = msgs::GetPropertiesResponse::FAILURE};
+  }
+
+  auto shard_acc = shard_->Access(req.transaction_id);
+  auto dba = DbAccessor{&shard_acc};
+  const auto view = storage::v3::View::NEW;
+
+  auto collect_props = [](const VertexAccessor &acc, const std::vector<PropertyId> &props, View view,
+                          const std::optional<EdgeAccessor> &e_acc) mutable -> std::optional<GetPropElement> {
+    std::vector<TypedValue> properties;
+    std::vector<PropertyId> ids;
+    for (const auto &prop : props) {
+      ShardResult<PropertyValue> result{PropertyValue()};
+      if (e_acc) {
+        result = e_acc->GetProperty(prop, view);
+      } else {
+        result = acc.GetProperty(prop, view);
+      }
+      if (result.HasError() && result.GetError() == common::ErrorCode::NONEXISTENT_OBJECT) {
+        continue;
+      }
+      if (result.HasError()) {
+        spdlog::debug("Encountered an Error while trying to get a vertex property.");
+        return std::nullopt;
+      }
```

Comment:
Please use `CreateErrorResponse` to properly log and transform the error into an error which can be returned to the query engine.